### PR TITLE
Update the Helm Chart README with some new features and warnings

### PR DESCRIPTION
**Pull request by scholzj**
### Type of change

- Task

### Description

This PR updates the Helm Chart README with some new features and warnings:
* It adds auto-rebalancing and tiered storage to the list of notable supported features
* It adds warnings about Zoo and MM2 removals

(This README is shipped with the Helm Chart and is displayed in in various catalogs such as [ArtifactHub](https://artifacthub.io/packages/helm/strimzi/strimzi-kafka-operator) or used locally by Helm users.)

### Checklist

- [x] Update documentation

### CODE REVIEW

**Review by see-quick — APPROVED** [review]

**Review by ppatierno — APPROVED** [review]

**Review by im-konge — APPROVED** [review]

File: packaging/helm-charts/helm3/strimzi-kafka-operator/README.md [hunk 1]
```diff
@@ -5,12 +5,18 @@ Strimzi provides a way to run an [Apache Kafka®](https://kafka.apache.org) clus
...
-Kubernetes 1.23 and 1.24 are not supported anymore.
+* From Strimzi 0.44.0 on, we support only Kubernetes 1.25 and newer.
+  Kubernetes 1.23 and 1.24 are not supported anymore.
+* ZooKeeper support will be soon removed from Apache Kafka and Strimzi.
+  Currently, the last Strimzi version with ZooKeeper support is expected to be Strimzi 0.45.
+  Please plan your migration to KRaft (ZooKeeper-less Apache Kafka) accordingly.
+  Follow the [documentation](https://strimzi.io/docs/operators/latest/full/deploying.html#assembly-kraft-mode-str) for more details. 
+* Kafka Mirror Maker 1 support will be soon removed from Apache Kafka and Strimzi.
+  Currently, the last Strimzi version with Mirror Maker 1 support is expected to be Strimzi 0.45.
+  Please plan your migration to Mirror Maker 2 or other mirroring tool.
```
Comment (PaulRMellor) [inline packaging/helm-charts/helm3/strimzi-kafka-operator/README.md:17]:
```suggestion
  Please plan your migration to Mirror Maker 2 or another mirroring tool.
```

File: packaging/helm-charts/helm3/strimzi-kafka-operator/README.md [hunk 2]
```diff
@@ -21,14 +27,16 @@ cluster using the [Helm](https://helm.sh) package manager.
...
 
 * **Manages the Kafka Cluster** - Deploys and manages all of the components of this complex application, including dependencies like Apache ZooKeeper® that are traditionally hard to administer.
-* **KRaft support** - Allows running Apache Kafka clusters in the KRaft mode (without ZooKeeper). 
+* **KRaft support** - Allows running Apache Kafka clusters in the KRaft mode (without ZooKeeper).
 * **Includes Kafka Connect** - Allows for configuration of common data sources and sinks to move data into and out of the Kafka cluster.
 * **Topic Management** - Creates and manages Kafka Topics within the cluster.
 * **User Management** - Creates and manages Kafka Users within the cluster.
 * **Connector Management** - Creates and manages Kafka Connect connectors.
-* **Includes Kafka Mirror Maker 1 and 2** - Allows for mirroring data between different Apache Kafka® clusters.
+* **Includes Kafka Mirror Maker** - Allows for mirroring data between different Apache Kafka® clusters.
```
Comment (PaulRMellor) [inline packaging/helm-charts/helm3/strimzi-kafka-operator/README.md:35]:
```suggestion
* **Includes Kafka MirrorMaker** - Allows for mirroring data between different Apache Kafka® clusters.
```

File: packaging/helm-charts/helm3/strimzi-kafka-operator/README.md [hunk 3]
```diff
@@ -21,14 +27,16 @@ cluster using the [Helm](https://helm.sh) package manager.
...
 * **Includes HTTP Kafka Bridge** - Allows clients to send and receive messages through an Apache Kafka® cluster via the HTTP protocol.
 * **Includes Cruise Control** - Automates the process of balancing partitions across an Apache Kafka® cluster.
+* **Auto-rebalancing when scaling** - Automatically rebalance the Kafka cluster after a scale-up or before a scale-down.
+* **Tiered storage** - Allows to offload older and less critical data to a lower cost and performance storage tier (such as object storage).
```
Comment (PaulRMellor) [inline packaging/helm-charts/helm3/strimzi-kafka-operator/README.md:39]:
```suggestion
* **Tiered storage** - Offloads older, less critical data to a lower-cost, lower-performance storage tier, such as object storage.
```

**Review by PaulRMellor — APPROVED** [review]
Looks good, Couple of minor suggestions

**Pull request merged**